### DISCUSSION TITLE
Is there any way to bypass the Pulumi error on getting some unexisting resource?

### DISCUSSION THREAD

**Discussion by mate201** [pulumi/pulumi#7316] [Pulumi in Practice]
I'm trying to figure out if a specific CRD exists in the cluster, and if it's not, create it. Otherwise, don't do anything. I can "get" a Custom Resource, but the problem is that if it doesn't exist, Pulumi fails on a preview step.

**Comment by rawkode** [comment]
Hi @mate201,

I know this is a little late to be able to help you but I'm going to answer for anyone else that comes across this issue.

Currently, `.get()` does error out if the the resource doesn't exist.

This is a known and tracked issue and I'm going to try and raise the priority of this and get it looked at.

- https://github.com/pulumi/pulumi/issues/3364

I'll follow up here as soon as I know more.
